Objects/dictobject.c

```diff
@@ -5685,3 +5742,27 @@ uint32_t _PyDictKeys_GetVersionForCurrentState(PyDictKeysObject *dictkeys)
     dictkeys->dk_version = v;
     return v;
 }
+
+void
+PyDict_Watch(PyObject* dict)
+{
+    if (PyDict_Check(dict)) {
```

Comment:
consider raising an exception (type error?) if someone tries watching something that isn't a dict

Reply:
Sure, that seems reasonable; I'll make this change.